### PR TITLE
minimal memory region system.

### PR DESCRIPTION
**Pull request by tobowers**
see the tests for usage, but this adds very simple 'region' system which allows for easy swapping of one or more memories to a region, and a reordering based on region.

### CODE REVIEW

**Review by neilsonnn — APPROVED** [review]

**Comment by neilsonnn** [comment]
looks great

File: packages/core/src/WorkingMemory.ts [hunk 1]
```diff
@@ -309,6 +343,107 @@ export class WorkingMemory extends EventEmitter {
...
+   * 
+   * @param regionName - The name of the region from which to remove memories.
+   * @returns A new WorkingMemory instance with the memories removed from the specified region.
+   * 
+   * @example
+   * ```
+   * const newWorkingMemory = workingMemory.withoutRegion("greetings");
+   * ```
+   */
+  withoutRegion(regionName: string) {
```
Comment (kafischer) [inline packages/core/src/WorkingMemory.ts:436]: thinking through ghost - would you want "withoutRegions" plural?

`withoutRegions(regionNames: string | string[])`

**Review by kafischer — CHANGES_REQUESTED** [review]

File: packages/core/src/WorkingMemory.ts (hunk 1 above)
Comment (tobowers) [reply packages/core/src/WorkingMemory.ts]: fixorated.

**Review by dooart — APPROVED** [review]
very elegant api 💅🏼 nice job!

**Review by kafischer — APPROVED** [review]
this is so good

**Pull request merged**